<?xml version="1.0"?>
<launch>
<param name="use_sim_time" value="true"/>

  <node name="gplanner" pkg="grasp_planner" type="grasp_planner_node" output="screen">
    
    <param name="gripper_file" value="$(find grasp_planner)/cfg/yumi_cyl_03mm.cons"/>
    <!--param name="gripper_file" value="$(find grasp_planner)/cfg/dummy.cons"/-->
    <param name="map_frame_name" value="sdf_map_frame"/>
    <param name="use_tf" value="true"/>
    <param name="sdf_map_topic" value="/sdf_map"/>

    <!-- asus camera -->
    <param name="depth_info_topic_name" value="/camera/depth_registered/camera_info"/>
    <param name="depth_topic_name" value="/camera/depth_registered/image_raw"/> 
    <!-- kinect2 camera -->
    <param name="depth_topic_name2" value="/kinect2/hd/image_depth_rect"/>
    <param name="depth_info_topic_name2" value="/kinect2/hd/camera_info"/>

    <!-- SDF related parameters -->
    <!--param name="LoadVolume" value= "$(find grasp_planner)/cfg/example.vti" /-->
    <param name="LoadVolume" value= "$(find grasp_planner)/cfg/newmap.vti" />
    <param name="runTrackerFromVolume" value= "true" />
    <param name="GridSizeX" value= "190"/>
    <param name="GridSizeY" value= "80"/>
    <param name="GridSizeZ" value= "270"/>
    <param name="PositiveTruncationDistance" value= "0.03" />
    <param name="NegativeTruncationDistance" value= "-0.003" />
    <param name="RobustStatisticCoefficient" value= "0.01" />
    <param name="OutputVolume" value= "false" />
    <param name="InteractiveMode" value= "false" />
    <param name="MaximumRaycastSteps" value= "360" /> 
    <param name="OutputTriangles" value= "false" />

<!-- full list of SDF parameters and their default values -->
<!-- 
    !note: deprecated, taken from gripper constraint map
    <param name="CellSize" value= "0.01" />
    <param name="ImageWidth" value= "640"/>
    <param name="ImageHeight" value= "480"/>
   
    <param name="OutputTriangles" value= "false" />
    <param name="InteractiveMode" value= "true" />
    
    <param name="MaxWeight" value= "64.0" />
    <param name="CellSize" value= "0.02" />
    <param name="GridSizeX" value= "256" />
    <param name="GridSizeY" value= "256" />
    <param name="GridSizeZ" value= "256" />
    <param name="offsetX" value= "0.0" />
    <param name="offsetY" value= "0.0" />
    <param name="offsetZ" value= "-0.40" />
    <param name="PositiveTruncationDistance" value= "0.1" />
    <param name="NegativeTruncationDistance" value= "-0.04" />
    <param name="RobustStatisticCoefficient" value= "0.02" />
    <param name="Regularization" value= "0.01" />
    <param name="MinPoseChangeToFuseData" value= "0.01" />
    <param name="ConvergenceCondition"  value= "0.0001" />
    <param name="MaximumRaycastSteps" value= "12" /> 
    <param name="c_name" value= "camera"/>
 -->   
<!--
    The following values are default for kinect-like sensors.
    If the parameters ImageWidth and ImageHeight are set to QVGA or QQVGA resolution,
    these values will be rescaled automatically.
    If set manually, remember to change them when using QVGA or QQVGA image modes.
-->
<!-- 
    <param name="FocalLengthX" value= "520.0" />
    <param name="FocalLengthY" value= "520.0" />
    <param name="CenterPointX" value= "319.5" />
    <param name="CenterPointY" value= "239.5" /> 
-->
  </node>
  <node pkg="tf" type="static_transform_publisher" name="roipub"
         args="0.32 0 0.22 -0.707 0 0 0.707 world sdf_map_frame 50" />

</launch>
